nixos/modules/services/mail/dovecot.nix

```
@@ -473,6 +468,17 @@ in
         description = "Programs available for use by the vnd.dovecot.pipe extension";
         type = types.listOf types.path;
       };
+
+      extraPluginSettings = mkOption {
+        type = types.attrsOf types.anything;
+        default = {};
+        example = literalExpression ''
+          {
+            sieve = "file:~/sieve;active=~/.dovecot.sieve";
+          }
+        '';
+        description = "Any extra settings to the dovecot2's sieve plugin, e.g. `sieve`, `sieve_default`, etc.";
```

```suggestion
        description = "Any extra plugin settings, e.g. `sieve`, `sieve_default`, etc.";
```

From what I can tell, the plugin block receives settings for any plugin, and I think it doesn't even matter if you put settings for different plugins into different plugin blocks or mix them. It would probably be better to move this out of sieve and make it "general purpose".

---

Hmm, sounds good.

---

I tried something, let me know, @2xsaiko, it's ideal to land this quickly otherwise we will break a lot of Dovecot2 setup out there :).